records_mover/db/bigquery/loader.py

```diff
@@ -19,19 +19,41 @@
 import logging
 from ..loader import LoaderFromFileobj
 from ..errors import NoTemporaryBucketConfiguration
+from ...check_db_conn_engine import check_db_conn_engine
 
 logger = logging.getLogger(__name__)
 
 
 class BigQueryLoader(LoaderFromFileobj):
     def __init__(self,
-                 db: Union[sqlalchemy.engine.Connection, sqlalchemy.engine.Engine],
+                 db: Optional[Union[sqlalchemy.engine.Connection, sqlalchemy.engine.Engine]],
                  url_resolver: UrlResolver,
-                 gcs_temp_base_loc: Optional[BaseDirectoryUrl])\
-            -> None:
+                 gcs_temp_base_loc: Optional[BaseDirectoryUrl],
+                 db_conn: Optional[sqlalchemy.engine.Connection] = None,
+                 db_engine: Optional[sqlalchemy.engine.Engine] = None) -> None:
+        db, db_conn, db_engine = check_db_conn_engine(db=db, db_conn=db_conn, db_engine=db_engine)
         self.db = db
+        self._db_conn = db_conn
+        self.db_engine = db_engine
         self.url_resolver = url_resolver
         self.gcs_temp_base_loc = gcs_temp_base_loc
+        self.conn_opened_here = False
+
+    def get_db_conn(self) -> sqlalchemy.engine.Connection:
```

Comment:
I see a lot of loaders that have this same pattern of a `get_db_conn` method and setter that does the same thing. Have you thought about extracting this into some shared base object?

Personally, I'm okay with limiting the amount of logic dumped into inherited base classes because I find the resulting code harder to follow, and this is relatively clear. 

But I wanted to bring it up as an option to be considered

Reply:
One advantage of extracting the logic could be when we move to change the db/dbconn/dbengine behavior again, it might make it simpler to update and test if its in fewer places

Reply:
Totally agree but, to your above point, I really try and shy away for very abstract inheritance as much as possible and this wasn't a place where I saw an easy way to implement a good inheritance pattern.

I always try to go with the `composition better than inheritance` mindset but since we're dealing with a property here, I didn't see an easy way to do that but happy to try and brainstorm with you!

Reply:
Alright @Brunope what do you think about this:
https://github.com/bluelabsio/records-mover/pull/255